### PR TITLE
Workflow names + tags + linting

### PR DESCRIPTION
**Pull request by bedroesb**
When I installed all the workflows of GTN I saw that there was no consistency in the workflow naming and that tags helped a lot in organizing the huge list of workflows.

- improved tool install in docker scripts (retry if tool install fails)
- standardized naming in workflows
- tagging of workflows with their topic
- linting for workflows without topic tag
- linting for workflows using tools from the testtoolshed #1663 
- disabled `CTB Unittest workflow` workflow because it contained testtools -> linting failed as expected
- updated contributing tutorial to add the topic tag to the new workflow

### CODE REVIEW

**Comment by hexylena** [comment]
I like this! That will be quite nice for workflow consumers.

Would you be willing to add a linting step to this PR? Just a quick bash script to check that workflows are tagged according to their topic?

**Comment by bedroesb** [comment]
@hexylena yes I could do that! Where do I have to add this bash script to?

**Comment by hexylena** [comment]
@bedroesb just add to bin, maybe named `validate-workflow-tags` and then update the Makefile to add it under the `check-workflow` target with the `validate-json` script that you can use as inspiration

File: topics/ecology/tutorials/de-novo-rad-seq/workflows/Galaxy-Workflow-de-novo-rad-seq.ga [hunk 1]
```diff
@@ -628,7 +628,7 @@
           "type": "tabular"
         },
         {
-          "name": "out_fastp\u0125ase",
+          "name": "out_fastpĥase",
```
Comment (bedroesb) [inline topics/ecology/tutorials/de-novo-rad-seq/workflows/Galaxy-Workflow-de-novo-rad-seq.ga]: is this non ascii character on purpose? `out_fastpĥase` There are 2 others.

Comment (bedroesb) [reply topics/ecology/tutorials/de-novo-rad-seq/workflows/Galaxy-Workflow-de-novo-rad-seq.ga]: I came across this while re-dumping the json, but changed it in a later commit back to its original

Comment (hexylena) [reply topics/ecology/tutorials/de-novo-rad-seq/workflows/Galaxy-Workflow-de-novo-rad-seq.ga]: yeah that's correct for that version: https://github.com/galaxyproject/tools-iuc/commit/0cdd1b2639fc9c834ba0804dcafb16391006ff6b

**Comment by bedroesb** [comment]
Ok I think this will work!

File: Makefile [hunk 2]
```diff
@@ -94,6 +94,11 @@ check-workflows: build ## validate Workflows
 		bash bin/validate-json.sh
 .PHONY: check-workflows
 
+check-workflow-tags: build ## validate Workflow tags
+	$(ACTIVATE_ENV) && \
+		bash bin/validate-workflow-tags.sh
```
Comment (shiltemann) [inline Makefile]: if you move your check to the `check-workflows` target above it, it will be included in the travis linting

Comment (bedroesb) [reply Makefile]: @shiltemann like:

```Makefile
check-workflows: build ## validate Workflows
	$(ACTIVATE_ENV) && \
		bash bin/validate-json.sh
		bash bin/validate-workflow-tags.sh
.PHONY: check-workflows
```

or
```Makefile
check-workflows: build ## validate Workflows
	$(ACTIVATE_ENV) && \
		bash bin/validate-json.sh
	$(ACTIVATE_ENV) && \
		bash bin/validate-workflow-tags.sh
.PHONY: check-workflows
```

(is my first time using a Makefile)

**Comment by bedroesb** [comment]
@hexylena since I am iterating over the workflows to check for the tags anyway, I also added a `tool from testtoolshed` test in the same script, as suggested/hinted   :)

**Comment by bedroesb** [comment]
TO DO: adding the recommendation to add the tag to the workflow to the contributing tutorial

**Pull request merged**